src/coreclr/jit/emitarm64.cpp

```diff
@@ -5844,6 +5844,12 @@ void emitter::emitIns_R_R_I(instruction     ins,
                 return;
             }
 
+            if ((reg1 == reg2) && (EA_SIZE(attr) == EA_PTRSIZE) && emitComp->opts.OptimizationEnabled() &&
```

Comment:
This doesn't necessarily have to be `EA_PTRSIZE`, right? Rather, we just need the offset to be a multiple of `8` in range (looks to be in the range of `-4096` to `4032`, inclusive)?

-- Not something I think that needs to be handled in this PR, but rather that might be a possible future improvement on top.

Reply:
Looks like it depends on the instruction. Some are multiples of 4/8/16 and some might be raw offsets. The ranges vary based on instruction too

Reply:
It has to be `EA_PTRSIZE` -- the register that is written is the address that was used for the load, so it is always pointer sized.

The offset is an unscaled 9-bit signed immediate for the (single register) loads that support the write-back. So -256 to 255 is supported for the combined form.

Reply:
> It has to be EA_PTRSIZE -- the register that is written is the address that was used for the load, so it is always pointer sized.

👍 

> The offset is an unscaled 9-bit signed immediate for the (single register) loads that support the write-back. So -256 to 255 is supported for the combined form.

Yeah, I was looking at the wrong instruction here. There's a few different `post-indexing` forms. For `ldp`, as an example, it's:
> For the 32-bit post-index and 32-bit pre-index variant: is the signed immediate byte offset, a multiple of 4 in the range -256 to 252, encoded in the "imm7" field as <imm>/4.
> For the 64-bit post-index and 64-bit pre-index variant: is the signed immediate byte offset, a multiple of 8 in the range -512 to 504, encoded in the "imm7" field as <imm>/8.

There's then `ldapr` which is fixed `4` or `8`, `ldiapp` which is fixed `8` or `16`, and `ldpsw` which has the `... a multiple of 4 in the range -256 to 252 ...` text


Seems we're not handling these ones in this PR, which is fine, just had a mental disconnect due to the differences between them 😄

Reply:
Yeah, some of those forms we could definitely handle in the future. One complication is that some of those forms allow redefining the GC ness of up to 3 registers which `instrDesc` does not support today, so we would need to expand it in some way.